File: src/core/python/core/model/demandPoint.py
import struct


class DemandPoint:
    """
    A class to represen demand point, i.e., the location and the number of
    people in a city that serves as demand.
    """

    def __init__(self, demandPointId: int, shortName: str, longName: str,
                 xCoordinate: float, yCoordinate: float, demand: int):
        """
        Constructur of a demand point.

        :param demandPointId    id of the demand point
        :param shortName        short name of the demand point
        :param longName         long name of the demand point
        :param xCoordinate      x coordniate of the demand point
        :param yCoordinate      y coordinate of the demand point
        :param demand           demand at the demand point
        """

        self.demandPointId = demandPointId
        self.shortName = shortName
        self.longName = longName
        self.xCoordinate = xCoordinate
        self.yCoordinate = yCoordinate
        self.demand = demand


    def getId(self) -> int:
        """
        Get the id of the demand point.
        :return     id of the demand point.
        """
        return self.demandPointId


    def getShortName(self) -> str:
        """
        Get the short name of the demand point
        :return  the short name of the demand point
        """
        return self.shortName


    def getLongName(self) -> str:
        """
        Get the long name of the demand point
        :return  the long name of the demand point
        """
        return self.longName


    def getxCoordinate(self) -> float:
        """
        Get the x coordinate of the demand point.
        :return  the x coordinate of the demand point
        """
        return self.xCoordinate


    def getyCoordinate(self) -> float:
        """
        Get the y coordinate of the demand point.
        :return  the y coordinate of the demand point
        """
        return self.yCoordinate


    def getDemand(self) -> int:
        """
        Get the demand at the demand point.
        :return     the demand
        """
        return self.demand

    def __str__(self) -> str:
        return "Demand (" + ", ".join(self.toCsvStrings()) + ")"

    def __eq__(self, other) -> bool:
        if not isinstance(other, DemandPoint):
            return False
        return (other.getId() == self.getId()
                and math.isclose(other.getxCoordinate(), self.getxCoordinate)
                and math.isclose(other.getyCoordinate(), self.getyCoordinate)
                and other.getDemand() == self.getDemand()
                and other.getShortName() == self.getShortName()
                and other.getLongName() == self.getLongName())

    def __ne__(self, other) -> bool:
        return not self == other

    def __hash__(self) -> int:
        return hash((self.demandPointId, self.shortName, self.longName, self.xCoordinate, self.yCoordinate, self.demand))

    def toCsvStrings(self) -> [str]:
        """
        Create a csv array containing the values needed for a demnad file in LinTim
        format.
        :return     a representation of this demandin LinTim format.
        """
        return [str(self.getId()),
                self.shortName,
                self.longName,
                str(self.getxCoordinate()),
                str(self.getyCoordinate()),
                str(self.getDemand())]

File: src/core/java/src/net/lintim/model/DemandPoint.java
package net.lintim.model;

import java.util.Arrays;

/**
 * A class to represent demand point, i.e., the location and the number of people in a city that serves
 * as demand.
 */
public class DemandPoint {
    protected int id;
    protected String shortName;
    protected String longName;
    protected double xCoordinate;
    protected double yCoordinate;
    protected int demand;

    /**
     * Constructor of a demand point.
     *
     * @param id          id of the demand point
     * @param shortName   short name of the demand point
     * @param longName    long name of the demand point
     * @param xCoordinate x coordinate of the demand point
     * @param yCoordinate y coordinate of the demand point
     * @param demand      demand at the demand point
     */
    public DemandPoint(int id, String shortName, String longName, double xCoordinate, double yCoordinate, int demand) {
        this.id = id;
        this.shortName = shortName;
        this.longName = longName;
        this.xCoordinate = xCoordinate;
        this.yCoordinate = yCoordinate;
        this.demand = demand;
    }

    /**
     * Gets the id of the demand point.
     *
     * @return id of the demand point
     */
    public int getId() {
        return id;
    }

    /**
     * Gets the short name of the demand point.
     *
     * @return the short name of the demand point
     */
    public String getShortName() {
        return shortName;
    }

    /**
     * Gets the long name of the demand point.
     *
     * @return the long name of the demand point
     */
    public String getLongName() {
        return longName;
    }

    /**
     * Gets the x coordinate of the demand point.
     *
     * @return the x coordinate of the demand point.
     */
    public double getxCoordinate() {
        return xCoordinate;
    }

    /**
     * Gets the y coordinate of the demand point.
     *
     * @return the y coordinate of the demand point
     */
    public double getyCoordinate() {
        return yCoordinate;
    }

    /**
     * Gets the demand at the demand point.
     *
     * @return the demand
     */
    public int getDemand() {
        return demand;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        DemandPoint that = (DemandPoint) o;

        if (getId() != that.getId()) return false;
        if (Double.compare(that.getxCoordinate(), getxCoordinate()) != 0) return false;
        if (Double.compare(that.getyCoordinate(), getyCoordinate()) != 0) return false;
        if (getDemand() != that.getDemand()) return false;
        if (getShortName() != null ? !getShortName().equals(that.getShortName()) : that.getShortName() != null)
            return false;
        return getLongName() != null ? getLongName().equals(that.getLongName()) : that.getLongName() == null;
    }

    @Override
    public int hashCode() {
        int result;
        long temp;
        result = getId();
        result = 31 * result + (getShortName() != null ? getShortName().hashCode() : 0);
        result = 31 * result + (getLongName() != null ? getLongName().hashCode() : 0);
        temp = Double.doubleToLongBits(getxCoordinate());
        result = 31 * result + (int) (temp ^ (temp >>> 32));
        temp = Double.doubleToLongBits(getyCoordinate());
        result = 31 * result + (int) (temp ^ (temp >>> 32));
        result = 31 * result + getDemand();
        return result;
    }

    /**
     * Create a csv array containing the values needed for a demand file in LinTim format
     * @return a representation of this demand in LinTim format
     */
    public String[] toCsvString(){
        return new String[]{
            String.valueOf(id),
            shortName,
            longName,
            String.valueOf(xCoordinate),
            String.valueOf(yCoordinate),
            String.valueOf(demand)
        };
    }

    @Override
    public String toString() {
        return "Demand " + Arrays.toString(toCsvString());
    }
}
